foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -112,6 +111,25 @@ def handle_repeated_failure(job) -> None:
     # during early testing stages.
     logger.warn("%s #%d failed %d times!!!", job.__class__.__name__, job.id, MAX_NUM_RETRIES + 1)
 
+def get_max_downloader_jobs(window=datetime.timedelta(minutes=2), nomad_client=None):
+    """
+    Fetches the desired maximum number of downloader jobs available based on the cluster size.
+    If this has been calculated recently, returns a cached value, else it will calculate it fresh every `window`.
+    """
+    global MAX_TOTAL_DOWNLOADER_JOBS
+    global TIME_OF_LAST_SIZE_CHECK
+
+    if (timezone.now() - TIME_OF_LAST_SIZE_CHECK > window):
+        # Assuming they're all similar to an X1, give 50 downloaders per node.
+        if not nomad_client:
+            nomad_host = get_env_variable("NOMAD_HOST")
+            nomad_port = get_env_variable("NOMAD_PORT", "4646")
+            nomad_client = Nomad(nomad_host, port=int(nomad_port), timeout=30)
+
+        MAX_TOTAL_DOWNLOADER_JOBS = len(nomad_client.nodes) * 50
```

Comment:
Our cluster can have 10-26 nodes, which means this value will generally end up being 500-1300. Right now we have this value hardcoded to be 400 and our database has been sitting 90-98% CPU utilization. Before we deploy this, I think we should do a load test with the value at 1300 to make sure that our database isn't going to go down if we ever end up with 26 small nodes.

Reply:
Ok based on our discussion, we're going to put a cap of 1000 no matter the cluster size. I guess the one last thing is that we should decrease the count of nodes by 1 because the smasher instance doesn't run downloader jobs.